homeassistant/components/fronius/sensor.py

```diff
@@ -146,24 +146,30 @@ def data(self):
         """Return the state attributes."""
         return self._fetched
 
+    @property
+    def available(self):
+        """Is the fronius device active?"""
+        return self._available
+
     async def async_update(self):
         """Retrieve and update latest state."""
         values = {}
         try:
             values = await self._update()
-            self._connection_failure = False  # reset connection failure
         except ConnectionError:
             # fronius devices are often powered by self-produced solar energy
             # and henced turned off at night.
             # Therefore we will not print multiple errors when connection fails
-            if not self._connection_failure:
-                self._connection_failure = True
+            if self._available:
+                self._available = False
                 _LOGGER.error("Failed to update: connection error")
         except ValueError:
             _LOGGER.error(
                 "Failed to update: invalid response returned."
                 "Maybe the configured device is not supported"
             )
+        else:
+            self._available = True  # reset connection failure
 
         if not values:
```

Comment:
If we `return` in the except blocks we can remove this check and the `else:` block and just set available to True in the normal flow.

Reply:
Thats a good observation!